tests/serde_tx_schema.rs

```diff
@@ -0,0 +1,3434 @@
+          "format": "uint64",
+          "minimum": 0.0
+        },
+        "lo": {
+          "type": "integer",
+          "format": "uint64",
+          "minimum": 0.0
+        }
+      }
+    },
+    "UInt256Parts": {
+      "description": "UInt256Parts is an XDR Struct defines as:\n\n```text struct UInt256Parts { uint64 hi_hi; uint64 hi_lo; uint64 lo_hi; uint64 lo_lo; }; ```",
+      "type": "object",
+      "required": [
+        "hi_hi",
+        "hi_lo",
+        "lo_hi",
+        "lo_lo"
+      ],
+      "properties": {
+        "hi_hi": {
+          "type": "integer",
+          "format": "uint64",
+          "minimum": 0.0
+        },
+        "hi_lo": {
+          "type": "integer",
+          "format": "uint64",
+          "minimum": 0.0
+        },
+        "lo_hi": {
+          "type": "integer",
+          "format": "uint64",
+          "minimum": 0.0
+        },
+        "lo_lo": {
+          "type": "integer",
+          "format": "uint64",
+          "minimum": 0.0
+        }
+      }
+    },
+    "Uint256": {
+      "description": "Uint256 is an XDR Typedef defines as:\n\n```text typedef opaque uint256[32]; ```",
+      "type": "array",
+      "items": {
+        "type": "integer",
+        "format": "uint8",
+        "minimum": 0.0
+      },
+      "maxItems": 32,
+      "minItems": 32
+    },
+    "VecM_for_Asset_and_5": {
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/Asset"
+      }
+    },
+    "VecM_for_ClaimPredicate_and_2": {
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/ClaimPredicate"
+      }
+    },
+    "VecM_for_Claimant_and_10": {
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/Claimant"
+      }
+    },
+    "VecM_for_DecoratedSignature_and_20": {
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/DecoratedSignature"
+      }
+    },
+    "VecM_for_LedgerKey_and_4294967295": {
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/LedgerKey"
+      }
+    },
+    "VecM_for_Operation_and_100": {
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/Operation"
+      }
+    },
+    "VecM_for_ScMapEntry_and_4294967295": {
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/ScMapEntry"
+      }
+    },
+    "VecM_for_ScVal_and_4294967295": {
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/ScVal"
+      }
+    },
+    "VecM_for_SignerKey_and_2": {
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/SignerKey"
+      }
+    },
+    "VecM_for_SorobanAuthorizationEntry_and_4294967295": {
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/SorobanAuthorizationEntry"
+      }
+    },
+    "VecM_for_SorobanAuthorizedInvocation_and_4294967295": {
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/SorobanAuthorizedInvocation"
+      }
+    }
+  }
+}"##,
```

Comment:
Here appears a JSON schema for the Transaction Envelope. 😬 cc @willemneal

Reply:
Hmm I wonder why the numbers at the end. I'll take a closer look.

Reply:
So it seems that types like `VecM_for_SorobanAuthorizationEntry_and_4294967295` are because of how it handles generics
```rust
pub struct VecM<T, const MAX: u32 = { u32::MAX }>(Vec<T>);
```

I'll look into how we can make a special case for `VecM` to be treated like a generic array with length bounds.


```json
 "SorobanAuthorizedInvocation": {
      "description": "SorobanAuthorizedInvocation is an XDR Struct defines as:\n\n```text struct SorobanAuthorizedInvocation { SorobanAuthorizedFunction function; SorobanAuthorizedInvocation subInvocations<>; }; ```",
      "type": "object",
      "required": [
        "function",
        "sub_invocations"
      ],
      "properties": {
        "function": {
          "$ref": "#/definitions/SorobanAuthorizedFunction"
        },
        "sub_invocations": {
          "$ref": "#/definitions/VecM_for_SorobanAuthorizedInvocation_and_4294967295"
        }
      }
    },
```

Should be

Reply:
Is the problem related to this issue on the schemars repo?
- https://github.com/GREsau/schemars/issues/111

Are you able to articulate the problem and post about it in that or a new issue over there?

Reply:
I'm still digging into it. I started by manually implementing `VecM`. I'm making a PR to this branch.

Reply:
Here: https://github.com/stellar/rs-stellar-xdr/pull/349

Reply:
I guess we don't really need custom impls, because the issue is only with types that have generics, and the only types that do are `VecM`, `BytesM`, `StringM`, and `Frame`, which are all stock types not generated. So I should be able to copy #349 and into xdrgen.

Reply:
I suggest putting the `impl`s in a non-generated file in this crate. It will help debugging and testing.